src/pkg/util/nvidiautils/nvidiautils.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package nvidiautils
+
+import (
+	"bufio"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// GetNvidiaBindPath returns a string array consisting of filepaths of nvidia
+// related files to be added to the BindPaths
+func GetNvidiaBindPath(abspath string) []string {
+	var strArray []string
+	var bindArray []string
+	var searchArray []string
+	var commentID = regexp.MustCompile(`#`)
+	var soID = regexp.MustCompile(".so")
+
+	// use nvidia-container-cli (if present)
+	command, err := exec.LookPath("nvidia-container-cli")
+	if err == nil {
+		cmd := exec.Command(command, "list", "--binaries", "--ipcs", "--libraries")
+		out, err := cmd.Output()
+		if err == nil {
+
+			for _, line := range strings.Split(string(out), "\n") {
+				val := soID.FindString(line) // this will disallow binaries (non .so files)
+				if val != "" {               // contains .so
+					if line != "" {
+						// extract the filename from the path
+						fileNames := strings.SplitAfter(line, "/")
+						fileName := fileNames[len(fileNames)-1]
+
+						testString := line + ":/.singularity.d/libs/" + fileName
+						bindArray = append(bindArray, testString)
+						strArray = append(strArray, fileName)
+					}
+				} else { // binary executable
+					bindArray = append(bindArray, line)
+					strArray = append(strArray, line)
+				}
+			}
+		}
+	}
+
+	cliEntries := strings.Join(strArray, " ") // save away for later comparison check (disallow duplicates)
+
+	// grab the entries in nvliblist.conf file
+	// use ldconfig to pattern match from ld.so.cache
+	newpath, err := filepath.Glob(abspath + "/nvliblist.conf")
+	if err == nil {
+		for _, filename := range newpath {
+
+			file, err := os.Open(filename)
+			if err == nil {
+				scanner := bufio.NewScanner(file)
+				for scanner.Scan() {
+					line := scanner.Text()
+					val := commentID.FindString(line)
+					if val == "" && line != "" {
+						searchArray = append(searchArray, line)
+					}
+				}
+				file.Close()
+			}
+		}
+	}
+
+	// walk thru the ldconfig output and add entries which contain the filenames located in
+	// the nvliblist.conf file (ldconfig filenames are full filepaths)
+	var searchFileName string
+	command, err = exec.LookPath("ldconfig")
+	if err == nil {
```

Comment:
```
if err != nil {
    return nil, fmt.Errorf("ldconfig not in path: %v", err)
}
```

Reply:
I agree with this though I think it should just spit a warning and then the function should return.